lib/src/components/text_field.dart

```diff
@@ -95,31 +176,45 @@ class ArDriveTextFieldState extends State<ArDriveTextField> {
             maxLength: widget.maxLength,
             focusNode: widget.focusNode,
             key: widget.key,
-            obscureText: widget.obscureText,
+            textInputAction: widget.textInputAction,
+            obscureText: _isObscureText,
             style: ArDriveTypography.body.inputLargeRegular(
               color: ArDriveTheme.of(context).themeData.colors.themeInputText,
             ),
-            autovalidateMode: AutovalidateMode.always,
+            autovalidateMode: widget.autovalidateMode,
             onChanged: (text) {
-              setState(
-                () {
-                  final textIsValid = widget.validator?.call(text);
-
-                  if (text.isEmpty) {
-                    textFieldState = TextFieldState.focused;
-                  } else if (textIsValid != null && !textIsValid) {
-                    textFieldState = TextFieldState.error;
-                  } else if (textIsValid != null && textIsValid) {
-                    textFieldState = TextFieldState.success;
-                  }
-
-                  widget.onChanged?.call(text);
-                },
-              );
+              validate(text: text);
+              widget.onChanged?.call(text);
             },
             autofillHints: widget.autofillHints,
             enabled: widget.isEnabled,
             decoration: InputDecoration(
+              suffix: GestureDetector(
+                onTap: () {
+                  setState(() {
+                    _obscurityToggle = !_obscurityToggle;
+                    _isObscureText = !_isObscureText;
```

Comment:
Do these have any difference in the meaning? If not, then consider consolidating them

Reply:
They don't. Consolodating them!